### PR TITLE
Add Reloading Shaders

### PR DESCRIPTION
**Pull request by OverlordsIII**
Adds a reloading shaders option for shaderpacks. By default, it is toggled to `R`

### CODE REVIEW

**Comment by OverlordsIII** [comment]
I'm not sure why github actions are not even running, could that be because of the new branch change?

File: src/main/java/net/coderbot/iris/Iris.java [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
+import net.minecraft.util.Formatting;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.lwjgl.glfw.GLFW;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/Iris.java:27]: Please don't reorder all of the imports unnecessarily.

Importing this code style should help:

```xml
<code_scheme name="Default" version="173">
  <option name="OTHER_INDENT_OPTIONS">
    <value>
      <option name="USE_TAB_CHARACTER" value="true" />
    </value>
  </option>
  <JavaCodeStyleSettings>
    <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
    <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
    <option name="PACKAGES_TO_USE_IMPORT_ON_DEMAND">
      <value />
    </option>
    <option name="IMPORT_LAYOUT_TABLE">
      <value>
        <package name="" withSubpackages="true" static="true" />
        <emptyLine />
        <package name="java" withSubpackages="true" static="false" />
        <emptyLine />
        <package name="javax" withSubpackages="true" static="false" />
        <emptyLine />
        <package name="" withSubpackages="true" static="false" />
        <emptyLine />
        <package name="net.minecraft" withSubpackages="true" static="false" />
        <emptyLine />
        <package name="net.fabricmc" withSubpackages="true" static="false" />
      </value>
    </option>
  </JavaCodeStyleSettings>
  <editorconfig>
    <option name="ENABLED" value="false" />
  </editorconfig>
  <codeStyleSettings language="Groovy">
    <indentOptions>
      <option name="USE_TAB_CHARACTER" value="true" />
    </indentOptions>
  </codeStyleSettings>
  <codeStyleSettings language="JAVA">
    <indentOptions>
      <option name="CONTINUATION_INDENT_SIZE" value="4" />
      <option name="USE_TAB_CHARACTER" value="true" />
    </indentOptions>
  </codeStyleSettings>
</code_scheme>
```

File: src/main/java/net/coderbot/iris/Iris.java [hunk 2]
```diff
@@ -63,8 +66,27 @@ public void onInitializeClient() {
 		if (currentPack == null) {
 			loadInternalShaderpack();
 		}
-
 		reloadKeybind = KeyBindingHelper.registerKeyBinding(new KeyBinding("iris.keybind.reload", InputUtil.Type.KEYSYM, GLFW.GLFW_KEY_R, "iris.keybinds"));
+		ClientTickEvents.END_CLIENT_TICK.register(minecraftClient -> {
+			while (reloadKeybind.wasPressed()){
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/Iris.java:71]: Why `while`?

File: src/main/java/net/coderbot/iris/Iris.java [hunk 3]
```diff
@@ -90,7 +112,7 @@ private void loadExternalShaderpack(String name) {
 
 	private void loadInternalShaderpack() {
 		Path root = FabricLoader.getInstance().getModContainer("iris")
-			.orElseThrow(() -> new RuntimeException("Failed to get the mod container for Iris!")).getRootPath();
+				.orElseThrow(() -> new RuntimeException("Failed to get the mod container for Iris!")).getRootPath();
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/Iris.java:115]: No need to mess with the indentation here

File: src/main/java/net/coderbot/iris/Iris.java [hunk 4]
```diff
@@ -102,6 +124,24 @@ private void loadInternalShaderpack() {
...
+		if (!irisConfig.isInternal()) {
+			shaderPackRoot = shaderpacksDirectory.resolve(irisConfig.getShaderPackName());
+		} else {
+			shaderPackRoot = FabricLoader.getInstance().getModContainer("iris").orElseThrow(() -> new RuntimeException("Failed to get the mod container for Iris!"))
+					.getRootPath()
+					.resolve("shaders");
+		}
+		ShaderPack pack = new ShaderPack(shaderPackRoot.resolve("shaders"));
+		pipeline = new ShaderPipeline(pack);
+		currentPack = pack;
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/Iris.java:142]: I'm going to assume that this code is duplicated, I would prefer for it to not be duplicated

File: src/main/java/net/coderbot/iris/Iris.java [hunk 5]
```diff
@@ -122,6 +162,7 @@ public static CompositeRenderer getCompositeRenderer() {
 		return compositeRenderer;
 	}
 
+
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/Iris.java:165]:
```suggestion
```

**Review by coderbot16 — CHANGES_REQUESTED** [review]

**Comment by coderbot16** [comment]
Merged via https://github.com/IrisShaders/Iris/commit/967fe3e52b1faa2dfc601046577efc390506b820

**Pull request closed without merging**